packages/react-aria-components/src/useDragAndDrop.tsx

```diff
@@ -63,7 +63,13 @@ interface DropHooks {
   ListDropTargetDelegate: typeof ListDropTargetDelegate
 }
 
-export type DragAndDropHooks = DragHooks & DropHooks;
+export type DragAndDropHooks = DragHooks & DropHooks & {
+  /**
+   * Determines whether drag and drop functionality is disabled for the collection.
+   * If set to true, both dragging items out of and dropping items into the collection are disabled.
+   */
+  isDisabled?: boolean
+};
```

Comment:
Sure, an option to disable drag-only or drop-only might be useful. However, since this PR is getting bigger, it might be more appropriate to address it separately. what do you think?

Reply:
I agree, we can handle it separately. I will file a separate issue if/when this PR goes in!

Reply:
Ah actually, I think we can get rid of this `isDisabled` from the return type since the draggable/droppable collection state objects should already have `isDisabled` as a property already